src/Symfony/Component/HttpFoundation/IpUtils.php

```diff
@@ -64,9 +64,13 @@ public static function checkIp4($requestIp, $ip)
         if (false !== strpos($ip, '/')) {
             list($address, $netmask) = explode('/', $ip, 2);
 
-            if ($netmask < 1 || $netmask > 32) {
+            if ($netmask < 0 || $netmask > 32) {
                 return false;
             }
+
+            if ('0' === $netmask) {
```

Comment:
The only address allowed to have a /0 netmask is 0.0.0.0. So this check must be: `0 === $netmask && '0.0.0.0' === $address`. Please don't cast 0 as string, do `0`, don't `'0'`.